hadoop-hdds/framework/src/main/java/org/apache/hadoop/hdds/security/x509/certificate/authority/DefaultCAServer.java

```diff
@@ -224,16 +227,20 @@ private KeyPair getCAKeys() throws IOException {
   public Future<CertPath> requestCertificate(
       PKCS10CertificationRequest csr,
       CertificateApprover.ApprovalType approverType, NodeType role) {
-    LocalDate beginDate = LocalDate.now().atStartOfDay().toLocalDate();
-    LocalDateTime temp = LocalDateTime.of(beginDate, LocalTime.MIDNIGHT);
+    LocalDateTime beginDate;
+    if (!testSecureFlag) {
```

Comment:
We had a discussion about the testSecureFlag today, let me summarize what I think in a few points:
- it would be beneficial to have the same code for production and tests
- it does not really matter from either production or test perspective if we date back the start of the validity period to the start of the day, it is only important for both that the certificate is valid once it is used
- there does not seem to be an industry standard that says a certificate validity period should start at midnight, in fact thinking about it, it is possibly just because of tooling, google for example has random timestamps for start of validity

Based on this I would suggest to use simply LocalDateTime.now() as the beginDate for all cases, and then calculate the endDate just by adding the duration.

Reply:
@fapifta . Yeah, it seems there is no industry standard or common practice for the certificate start and end time rounding, then let's keep the same logic using LocalDateTime.now().